src/form/Input.js

```diff
@@ -18,8 +18,10 @@
     Input.prototype._class += " form_Input";
     Input.prototype.implements(IInput.prototype);
 
-    Input.prototype.publish("type", "text", "set", "Input type", ["textbox", "number", "checkbox", "button", "select", "textarea", "date", "html-color"]);
+    Input.prototype.publish("type", "text", "set", "Input type", ["html-color", "number", "checkbox", "button", "select", "textarea", "date", "text", "range", "search", "email", "time", "datetime"]);
```

Comment:
```html-color``` should be ```color``` Plus you may as well add the official list of items:
https://developer.mozilla.org/en/docs/Web/HTML/Element/Input